crates/compilers/src/lib.rs

```diff
@@ -882,6 +890,36 @@ fn rebase_path(base: &Path, path: &Path) -> PathBuf {
     new_path.to_slash_lossy().into_owned().into()
 }
 
+/// Utility function to apply a set of updates to provided sources.
+fn apply_updates(sources: &mut Sources, updates: Updates) {
+    for (path, source) in sources {
+        if let Some(updates) = updates.get(path) {
+            source.content = Arc::new(replace_source_content(
+                source.content.as_str(),
+                updates.iter().map(|(start, end, update)| ((*start..*end), update.as_str())),
+            ));
+        }
+    }
+}
+
+/// Utility function to change source content ranges with provided updates.
+fn replace_source_content<'a>(
```

Comment:
I think this assumes that the updates are sorted?

Reply:
yeah, I added comment to reflect this in https://github.com/foundry-rs/compilers/pull/252/commits/f4a109d53bfe97bba287335404c2f5cc20e7dbf6